packages/discord.js/typings/index.d.ts

```diff
@@ -4669,6 +4730,23 @@ export interface SelectMenuComponentOptionData {
   value: string;
 }
 
+export interface TextInputComponentData extends BaseComponentData {
+  customId: string;
+  style: TextInputStyle;
+  label: string;
+  minLength?: number;
+  maxLength?: number;
+  required?: boolean;
+  value?: string;
+  placeholder?: string;
+}
+
+export interface ModalData {
+  customId: string;
+  title: string;
+  components: ActionRowData<ModalActionRowComponentData>;
```

Comment:
```suggestion
  components: ActionRowData<ModalActionRowComponentData>[];
```
ignoring the custom_id casing, this looks like a required typing change for raw modal constructor input